### PR TITLE
Improve use of promises - Closes #1437

### LINKED ISSUE #1437: Improve use of promises
This sort of thing is everywhere in the code:

```
return Promise.reject(/* some message or data */);
```

This creates a maintenance nightmare, running into one of those, because such a reject does not carry any stack-tracing information.

In order to make such rejects traceable, the following needs to be done:

* Every manual reject must always wrap the message into the error object, to provide the call stack:

```js
return Promise.reject(new Error(/* some message or data */));
```

It will make it possible for `Bluebird` to trace rejection stack, after being explicitly enabled:

* Globally in the application enable long stack tracing supported by `Bluebird`: http://bluebirdjs.com/docs/api/promise.longstacktraces.html

```js
Promise.config({ 
    longStackTraces: true // Enable long stack traces
});
```

...among other things, just make sure not setting these for production, because long stack traces slow down the performance.

### PR DESCRIPTION
**Pull request by vitaly-t**
### What was the problem?

Use of promises is inconsistent with the standard, see #1437 for details.

### How did I fix it?

* Enabled `Long Stack Tracing` feature of Bluebird, to provide proper stack tracing whenever a reject occurs.
* Made sure error stack is injected into every reject

### How to test it?

See that every reject comes with detailed stack trace, pointing exactly at the line of where reject occurred.

### Review checklist

* The PR solves #1437
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: test/common/ws/server_process.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ WSServer.prototype.start = function() {
 
 		self.socketCluster.on('fail', () => {
 			self.stop();
-			reject();
+			reject(new Error());
 		});
```
Comment (nazarhussain) [inline test/common/ws/server_process.js]: Same here.

File: test/common/ws/server_master.js [hunk 2]
```diff
@@ -62,7 +62,7 @@ WSServerMaster.prototype.start = function() {
 
 		self.masterProcess.on('error', () => {
 			self.stop();
-			reject();
+			reject(new Error());
```
[the remark this answers is not in the document]
Comment (vitaly-t) [reply test/common/ws/server_master.js]: Sorry, I'm not clear on what you are suggesting that I should do here...

What I did here - added stack trace, so when this reject happens, one can see the stack of where it came from exactly. Having some sort of message is optional here. So if you want one, please formulate it, and I will add one.

File: test/common/ws/server_process.js (hunk 1 above)
Comment (vitaly-t) [reply test/common/ws/server_process.js]: Same here :)

File: test/common/ws/server_master.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (nazarhussain) [reply test/common/ws/server_master.js]: It should be: 


```
self.masterProcess.on('error', (error) => {
   self.stop();
   reject(error);
});
```

File: test/common/ws/server_process.js (hunk 1 above)
Comment (nazarhussain) [reply test/common/ws/server_process.js]: It should be: 


```
self.socketCluster.on('fail', (error) => {
   self.stop();
   reject(error);
});
```

File: db/index.js [hunk 3]
```diff
@@ -24,6 +24,12 @@ const monitor = require('pg-monitor');
 const pgpLib = require('pg-promise');
 const repos = require('./repos');
 
+const inTest = process.env.NODE_ENV === 'test';
+
+Promise.config({
+	longStackTraces: inTest, // Enable Long Stack Traces
+});
+
```
[the remark this answers is not in the document]
Comment (nazarhussain) [reply db/index.js]: @vitaly-t What about this?

[the remark this answers is not in the document]
Comment (vitaly-t) [reply db/index.js]: This is a global setting, needs to be done in just one place, as `require('bluebird')` returns a singleton.

**Comment by nazarhussain** [comment]
@vitaly-t The PR is  failing on Jenkins, can you please check.

**Pull request merged**